### PR TITLE
Fix testing tests command to allow passing test as extra arg

### PR DESCRIPTION
**Pull request by potiuk**
When providers have been moved in #42505 broke passing parameters when provider tests were passed as extra args of "testing tests" command of breeze.

Previously all tests were under "tests" folder and there was an exclusion to disable "All" tests when any test was passed as parameter. But after moving tests to "providers" this stopped working.

Additional exclusion needs to be added for "providers/tests" and "providers/tests_sdk/".

This PR also adds autocompletion for tests passed this way by setting the click type to Path for the extra args (but without the need for the Path to exist).

Also during this check it turned out that "All" tests are not working in the intended way - but this should not impact our CI only local runs.

Appropriate comment has been added and it's captured in https://github.com/apache/airflow/issues/42632

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/commands/testing_commands.py
Comment (ferruzzi) [inline dev/breeze/src/airflow_breeze/commands/testing_commands.py:884]: Just curious, maybe a regex match on `^[a-zA-Z_]*/?tests/$`

`Any combination of letters and underscores (including zero), followed by an optional slash, and ending with the exact string 'tests/'` should match all `tests` directories at most one subdirectory deep and maybe make this a little less brittle?

File: dev/breeze/src/airflow_breeze/commands/testing_commands.py
Comment (o-nikolas) [inline dev/breeze/src/airflow_breeze/commands/testing_commands.py:556]: Does making this a Path inhibit folks from using it for it's originally intended use, which was passing more pytest options/flags?

Comment (potiuk) [reply dev/breeze/src/airflow_breeze/commands/testing_commands.py:556]: No. It's ok. It does not have `exists=True` so it does not check if the path exists, and path_type= `str` avoids conversion of the string passed to `Path`, so things like `--system x` will both be passed verbatim - I tested it.

File: dev/breeze/src/airflow_breeze/commands/testing_commands.py
Comment (potiuk) [reply dev/breeze/src/airflow_breeze/commands/testing_commands.py:884]: This is really a workaround -  temporary solution to unblock @o-nikolas and @ambika-garg from running system tests in the way they are used to. 

The TODO: and https://github.com/apache/airflow/issues/42632 describe a more "acurate" and complete solution to it - we will have to split the tests into three different groups now and treat them separately and run (possibly) different commands for all three tests so this one is going to change anyway (and we can implement it better because now logic is spread across different places and it's very difficult to reason about - it was non-obvious at all for @ashb when doing #42505 tthat this one should be also changed, which means that we have to simplify the whole setup here - and splitting to "airflow", "providers" and "test_sdk" types of tests is a good opportunity to do it "properly".

**Review by ashb — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

File: dev/breeze/src/airflow_breeze/commands/testing_commands.py
Comment (o-nikolas) [reply dev/breeze/src/airflow_breeze/commands/testing_commands.py:884]: Do we need to add `tests/system` as well to this list? Are tests are still not passing, here is the command that gets run for a system test (when you use the `--system` flag, `tests/system` is the leading path that should be removed?).
```
Starting the tests with those pytest arguments: tests/system --verbosity=0 --strict-markers --durations=100 --maxfail=50 --color=yes --junitxml=/files/test_result-system-postgres.xml --timeouts-order=moi --setup-timeout=5400 --execution-timeout=5400 --teardown-timeout=5400 --disable-warnings -rfEX --ignore-glob=*/tests/integration --warning-output-path=/files/warnings-system-postgres.txt --ignore=helm_tests --ignore=providers/tests/cloudant --ignore=providers/tests/system/cloudant --ignore=providers/tests/integration/cloudant --no-cov --keep-env-variables --no-db-cleanup providers/tests/system/amazon/aws/example_athena.py --system amazon -s --color=no
ERROR: found no collectors for /opt/airflow/providers/tests/system/amazon/aws/example_athena.py
============================= test session starts ==============================
platform linux -- Python 3.9.20, pytest-8.3.3, pluggy-1.5.0
rootdir: /opt/airflow
configfile: pyproject.toml
plugins: rerunfailures-14.0, xdist-3.6.1, mock-3.14.0, time-machine-2.16.0, requests-mock-1.12.1, icdiff-0.9, instafail-0.5.0, custom-exit-code-0.3.0, cov-6.0.0, asyncio-0.24.0, timeouts-1.2.1, anyio-4.6.2.post1, kgb-7.1.1
asyncio: mode=strict, default_loop_scope=function
setup timeout: 5400.0s, execution timeout: 5400.0s, teardown timeout: 5400.0s
collected 0 items / 1 error
==================================== ERRORS ====================================
_______________________ ERROR collecting providers/tests _______________________
Defining 'pytest_plugins' in a non-top-level conftest is no longer supported:
It affects the entire test suite instead of just below the conftest as expected.
  /opt/airflow/providers/tests/conftest.py
Please move it to a top level conftest file at the rootdir:
  /opt/airflow
For more information, visit:
  https://docs.pytest.org/en/stable/deprecations.html#pytest-plugins-in-non-top-level-conftest-files
---------- generated xml file: /files/test_result-system-postgres.xml ----------
=========================== short test summary info ============================
ERROR providers/tests - Failed: Defining 'pytest_plugins' in a non-top-level conftest is no longer supported:
It affects the entire test suite instead of just below the conftest as expected.
  /opt/airflow/providers/tests/conftest.py
Please move it to a top level conftest file at the rootdir:
  /opt/airflow
For more information, visit:
  https://docs.pytest.org/en/stable/deprecations.html#pytest-plugins-in-non-top-level-conftest-files
========================= 1 warning, 1 error in 0.05s ==========================
```

Comment (potiuk) [reply dev/breeze/src/airflow_breeze/commands/testing_commands.py:884]: aaaaa... Right iit's the "--test-type System` that should be removed from your call. I think that should help - and the refactor we need to do should help there as well - we shoudl remove some of the old test types that are not used any more

**Pull request merged**
